source/_posts/2025-05-12-mcp-explained-without-hype-or-fluff.markdown

```diff
@@ -0,0 +1,160 @@
+---
+title: MCP explained without hype or fluff
+kind: article
+author: Atharva Raykar
+created_at: 2025-05-12 00:00:00 UTC
+layout: post
+---
+Model Context Protocol, like most protocols, solves the M ⨯ N integration problem by turning it into an M + N integration problem.
+
+An AI client application that speaks this protocol does not have to figure out how to fetch data or take actions specific to a platform.
+
+![](/images/blog/mcp.jpg)
+
+MCP may not make your AI smarter, or improve your product, but it will reduce the friction to integrate against other applications that already support MCP. This may or may not be important to you.
```

Comment:
> or improve your product

This is debatable, you could argue that having access to better context, tools etc via MCP can make the product better as opposed to the lack of it. We should skip this IMO

Reply:
Does this rewording clarify the intent better: 

"MCP may or may not make your AI smarter or improve your product"

The intent of that is that MCP is just a tool and depending on how one wields it, it may or may not improve the product

Reply:
I don't want to skip this though, I think when writing and promoting a hot new thing, it's important to talk about the limitations also and be realistic about it

Reply:
Yeah, the reworded version LGTM